### PR TITLE
chore(gitlab): allow images to be published directly from pipeline

### PR DESCRIPTION
**Pull request by duncanista**
# What?

Allows `datadog/lambda-extension` image to be built for `amd64` and `arm64` (and their alpine versions) from the GitLab pipeline.

At the same time, this built will be used for automatic releases.

# How?

Leveraging [DataDog/public-images](https://github.com/DataDog/public-images) to automate the process.

# Motivation

We needed it.
Also [SVLS-5140](https://datadoghq.atlassian.net/browse/SVLS-5140)

[SVLS-5140]: https://datadoghq.atlassian.net/browse/SVLS-5140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitlab/templates/lambda-extension.yaml.tpl [hunk 1]
```diff
@@ -48,6 +55,58 @@ build bottlecap ({{ $architecture.name }}):
...
+    - build bottlecap ({{ $architecture.name }}, alpine)
+  dependencies:
+    - build bottlecap ({{ $architecture.name }}, alpine)
+  variables:
+    ARCHITECTURE: {{ $architecture.name }}
+    ALPINE: 1
+  script:
+    - .gitlab/scripts/build_image.sh
+# Alpine end
+
```
Comment (duncanista) [inline .gitlab/templates/lambda-extension.yaml.tpl:109]: This is better than adding a new entry into `architectures.yaml`, way easier even.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -36,14 +39,17 @@ generator:
...
-    - gomplate --config .gitlab/config.yaml
   artifacts:
     paths:
       - .gitlab/pipeline-bottlecap.yaml
       - .gitlab/pipeline-go-agent.yaml
       - .gitlab/pipeline-lambda-extension.yaml
+    reports:
+      dotenv: .env
+  script:
+    - if [[ "$CI_COMMIT_TAG" =~ ^v[0-9]+$ ]]; then echo "VERSION=$(echo "${CI_COMMIT_TAG##*v}" | cut -d. -f2)" >> .env; fi
```
Comment (duncanista) [inline .gitlab-ci.yml:50]: Allows us to send a parsed version of the layer for the `publish images` job which requires it as an environment variable. This is the only way to send it before those jobs get generated.

Comment (alexgallotta) [reply .gitlab-ci.yml:50]: should the env file `export ENV`?

**Review by alexgallotta — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (duncanista) [reply .gitlab-ci.yml:50]: I'm not sure I understand the question

**Pull request merged**